arrayMethodsLab.js

```diff
@@ -0,0 +1,308 @@
+// ```
+console.log("Question 4: ")
+let array4 = [1, 1, 1, 1, 1, 2];
+const areAllTheElementsTheSame = (currentValue) => {
+    //console.log(currentValue)
+    return currentValue === array4[0]; //array4 is global
+}
+let output4 = array4.every(areAllTheElementsTheSame);
+console.log(output4);
+console.log("------------------------------------------------------------")
+// 5. Write a function that returns the sum of every element in an array.
+// ```
+// Input: [1,2,3,4,5]
+// Output: 15
+// ```
+console.log("Question 5: ")
+let array5 = [1, 2, 3, 4, 5];
+const adder = (total, currentElem) => {
+    return total + currentElem;
+}
+let sum = ( array5.reduce(adder, 0) );
+console.log(sum);
+console.log("------------------------------------------------------------")
+// 6. Write a function that finds largest number in an array.
+
+// ```
+// Input: [1,2,6,3,4,5]
+// Output: [6]
+// ```
+console.log("Question 6: ");
+let array6 = [1, 2, 6, 3, 4, 5]
+const largest = array6.reduce( (total, currentElem) => {
+    if(currentElem > total) {
+        return currentElem;
+    } else {
+        return total;
+    }
+})
+console.log(largest);
+console.log("------------------------------------------------------------");
+// 7. Write a function `countZeroes`, which takes an array of numbers as its argument and returns the amount of zeroes that occur in it.
+
+// ```
+// Input: [1,0,0,2,0,3,0,0,0]
+// Output: 6
+// ```
+console.log("Question 7: ");
+let array7 = [1, 0, 0, 2, 0, 3, 0, 0, 0];
+let counter7 = 0;
+array7.forEach(function(element) {
+    if(element === 0) {
+        counter7++;
+    }
+});
+console.log(counter7);
+console.log("------------------------------------------------------------");
+// 8. Write a function that doubles every element in an array.
+
+// ```
+// Input: [1,2,3,4]
+// Output: [2,4,6,8]
+// ```
+console.log("Question 8: ");
+let array8 = [1, 2, 3, 4];
+let output8 = array8.map( (currentElem) => {
+    return currentElem *= 2
+})
+console.log(output8);
+console.log("------------------------------------------------------------");
+// 9. Write a function that finds the sum of every element in an array doubled
+
+// ```
+// Input: [1,2,3,4]
+// Output: 20
+// ```
+console.log("Question 9: ");
+let array9 = [1, 2, 3, 4];
+const sum9 = (total, currentElem) => {
+    return total + currentElem;
+}
+let final = ( array9.reduce(sum9, 0) * 2 );
+console.log(final);
+console.log("------------------------------------------------------------");
+// 10. Write a function `elementDivisibleBy` with parameters `divisor` and `arr`.  Return a new array of every element of arr that can be evenly divided by divisor.
+
+// ```
+// Input: 5, [4,5,6,7,8,9,10,11,12,13,14,15]
+// Output: [5,10,15]
+// ```
+console.log("Question 10: ");
+let array10 = [4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15];
+const elementDivisibleBy = (divisor) => {
+    return divisor % 5 === 0;
+}
+let output10 = array10.filter(elementDivisibleBy)
+console.log(output10);
```

Comment:
This isn't correct. What two arguments should the function take in? We want to use the filter method within the function... what would the condition be for it to be reusable code?